packages/assertion-monitor/chains.ts

```diff
@@ -0,0 +1,63 @@
+import { ChildNetwork } from 'utils'
+import { Chain } from 'viem'
+import {
+  mainnet,
+  arbitrum,
+  arbitrumNova,
+  base,
+  sepolia,
+  holesky,
+  arbitrumSepolia,
+  baseSepolia,
+} from 'viem/chains'
+
+export const DEFAULT_TIMESPAN_SECONDS = 60 * 60 * 24 * 7 // 1 week
+
+export const supportedParentChains = [
+  mainnet,
+  arbitrum,
+  arbitrumNova,
+  base,
+
+  sepolia,
+  holesky,
+  arbitrumSepolia,
+  baseSepolia,
+]
+
+export const getChainFromId = (chainId: number): Chain => {
+  const chain = supportedParentChains.filter(chain => chain.id === chainId)
+  return chain[0] ?? null
+}
+
+export const getDefaultBlockRange = (chain: Chain): bigint => {
+  switch (chain) {
+    case mainnet:
+    case sepolia:
+    case holesky:
+      return BigInt(DEFAULT_TIMESPAN_SECONDS / 12)
+
+    case base:
+    case baseSepolia:
+      return BigInt(DEFAULT_TIMESPAN_SECONDS / 2)
+
+    case arbitrum:
+    case arbitrumNova:
+    case arbitrumSepolia:
+      return BigInt(DEFAULT_TIMESPAN_SECONDS / 12)
```

Comment:
Since block range will only be dependent on L1, can we just have base/baseSepolia as 1 case, and for others, return the same DEFAULT_TIMESPAN_SECONDS/12 ?

Reply:
I don't think that's the case here (addressed in [this comment too](https://github.com/OffchainLabs/arbitrum-monitoring/pull/25#discussion_r1715564113)). We are searching for logs in a range of blocks, so if the parent chain is an arbitrum chain, we should calculate a range of blocks according to Arbitrum's block speed (up to 4 blocks per second).

If we were doing a calculation based on block times, then we could be using Ethereum's block time (depending on what we are doing), but in this case we are searching for logs.

Reply:
Hmm.. A bit confusing but for batch poster monitor as well, we used L1 block time for checking batches - https://github.com/OffchainLabs/arbitrum-monitoring/blob/main/packages/batch-poster-monitor/chains.ts#L68

Compared to previous change - Now updating to `DEFAULT_TIMESPAN_SECONDS  * 4` for arbitrum chains is stalling the script a lot since now there are a ton of logs that are being processed.

@TucksonDev do you think there is a way we can optimize this?

Reply:
In the batch poster monitor, we are using `batchPostingTimeBounds` to calculate when we expect a batch poster to post the batches. And that calculation is based on the delaySeconds and delayBlocks variables of the SequencerInbox, which uses block.number, so it's based on Ethereum block numbers and times for any Arbitrum and Orbit chains. Basically, we are not using it to query logs, but to calculate block times, so it's ok to use 12 seconds.

The problem with the script now is that it's trying to query 7 days of logs, which is a lot (2.5M of blocks for Arbitrum chains). There are several solutions to this:
- Reduce the 7 day window
- Search iteratively from end to beginning of the range, and stop when we find a log
- Somehow detect if the script is searching too many logs for a specific chain and try to parallelize that process (so it doesn't affect processing the other chains). This one might be a bit tricky, and I'm not 100% sure about it.